### PR TITLE
Authentication pages and functionalities

### PR DESCRIPTION
**Pull request by yaswanthmyneni**
# Authentication pages, functionalities, and Toast

### CODE REVIEW

File: src/components/require-auth/RequireAuth.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { Navigate, useLocation } = require("react-router-dom");
+
+const RequireAuth = ({ children }) => {
+  let location = useLocation();
```
Comment (sri-rishi) [inline src/components/require-auth/RequireAuth.jsx]: Suggestion: you can use `const` variable here

**Review by sri-rishi — APPROVED** [review]
Code looks good, just some suggestion which I think can be done, else feel free to merge 

**When click on `Join Now` button at home page it's asking me to signup even though I am logged in **

image

File: src/context/authentication-context.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const [authState, authDispatch] = useReducer(authenticationReducer, {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+    user: {},
+  });
+
+  const value = { authState, authDispatch };
```
Comment (omkarpatke) [inline src/context/authentication-context.js:50]: suggestion: you can directly add the values no need to create variable.

**Review by omkarpatke — APPROVED** [review]
not found any required suggestions, variable naming is very good code is clean and readable you can merge after some changes.

File: src/context/authentication-context.js (hunk 2 above)
Comment (yaswanthmyneni) [reply src/context/authentication-context.js:50]: I think this way is more readable if we have to send more variables and functions to context.

**Pull request merged**